### PR TITLE
[Merge v3.0.0] into Develop

### PR DESCRIPTION
**Pull request by kcw-grunt**
- version bump

### CODE REVIEW

File: loafwallet/CardView.swift [hunk 1]
```diff
@@ -62,21 +70,29 @@ struct CardView: View {
                 Group {
                     AnimatedCardView(viewModel: animatedViewModel, isLoggedIn: $didCompleteLogin)
                         .frame(minWidth:0,
-                               maxWidth: didCompleteLogin ? geometry.size.width * 0.4 :  geometry.size.width * 0.6)
+                               maxWidth:
+                                didCompleteLogin ? geometry.size.width * 0.4 :
```
Comment (mosadialiou) [inline loafwallet/CardView.swift:74]: Just curious: did you do the line break yourself OR it's the formatter?

File: loafwallet/CardView.swift [hunk 2]
```diff
@@ -62,21 +70,29 @@ struct CardView: View {
...
                 
                 //MARK: - Login Textfields
                 Group {
                     
-                    TextField(S.Receive.emailButton, text: $loginModel.emailString) {  _ in
-                       cannotLogin = loginModel.emailString.count < 4
+                    TextField(S.Receive.emailButton,
+                              text: $loginModel.emailString)
+                        .onReceive(loginModel.$emailString) { currentEmail in
+                            if currentEmail.count < 4 ||  !registrationModel.isEmailValid(emailString: currentEmail) {
```
Comment (mosadialiou) [inline loafwallet/CardView.swift:85]: replace if else with:
`isEmailValid = currentEmail.count > 4 && registrationModel.isEmailValid(emailString: currentEmail)`

File: loafwallet/CardView.swift [hunk 3]
```diff
@@ -86,6 +102,14 @@ struct CardView: View {
                         if shouldShowPassword {
                             
                             TextField(S.Import.passwordPlaceholder.capitalized, text: $loginModel.passwordString)
+                                .onReceive(loginModel.$passwordString) { currentPassword in
+                                    if !registrationModel.isPasswordValid(passwordString:  currentPassword) {
```
Comment (mosadialiou) [inline loafwallet/CardView.swift:106]: replace `if else` with 

`isPasswordValid = registrationModel.isPasswordValid(passwordString:  currentPassword)`

File: loafwallet/CardView.swift [hunk 4]
```diff
@@ -96,6 +120,14 @@ struct CardView: View {
                         } else {
                             
                             SecureField(S.Import.passwordPlaceholder.capitalized, text: $loginModel.passwordString)
+                                .onReceive(loginModel.$passwordString) { currentPassword in
+                                    if !registrationModel.isPasswordValid(passwordString:  currentPassword) {
```
Comment (mosadialiou) [inline loafwallet/CardView.swift:124]: replace `if else` with

`isPasswordValid = registrationModel.isPasswordValid(passwordString:  currentPassword)`

File: loafwallet/RegistrationViewModel.swift [hunk 5]
```diff
@@ -101,19 +114,21 @@ class RegistrationViewModel: ObservableObject {
...
         switch dataType {
             case .genericString:
-                return isGenericStringValid(genericString: (data as! String))
+                return isGenericStringValid(genericString: dataString)
             case .email:
-                return isEmailValid(emailString: (data as! String))
+                return isEmailValid(emailString: dataString)
             case .country:
-                return (data as! String) == "US" ?  true : false
+                return  dataString == "US" ?  true : false
```
Comment (mosadialiou) [inline loafwallet/RegistrationViewModel.swift:125]: write just:

`dataString == "US"`

File: loafwallet/RegistrationViewModel.swift [hunk 6]
```diff
@@ -39,16 +39,29 @@ class RegistrationViewModel: ObservableObject {
...
+        guard let email = data["email"] as? String else { return }
+        guard let password = data["password"] as? String else { return }
+        guard let phone = data["phone"] as? String else { return }
+        guard let country = data["country"] as? String else { return }
+        guard let state = data["state"] as? String else { return }
+        guard let city = data["city"] as? String else { return }
+        guard let address1 = data["address1"] as? String else { return }
+        guard let zip = data["zip_code"] as? String else { return }
+        
+        if  self.isDataValid(dataType: .genericString, data: first) &&
```
Comment (mosadialiou) [inline loafwallet/RegistrationViewModel.swift:53]: as a general remark, `self.` is not needed. Just write the function's name.

**Review by mosadialiou — CHANGES_REQUESTED** [review]

**Comment by kcw-grunt** [comment]
This is stale...closing once v3.0.0 is approved.

File: loafwallet/CardView.swift (hunk 1 above)
Comment (kcw-grunt) [reply loafwallet/CardView.swift:74]: manual but the formatter will do the same.

File: loafwallet/CardView.swift (hunk 3 above)
Comment (kcw-grunt) [reply loafwallet/CardView.swift:106]: no need to set the variable...its changed by the model...

File: loafwallet/CardView.swift (hunk 4 above)
Comment (kcw-grunt) [reply loafwallet/CardView.swift:124]: no need to set the variable...its changed by the model...

File: loafwallet/RegistrationViewModel.swift (hunk 5 above)
Comment (kcw-grunt) [reply loafwallet/RegistrationViewModel.swift:125]: I really really like ternary operators...I need the gentle jog of of memory of operation.   Even though we can leave it shorter, this method has a good reminder of the ideal logic.  That's why I put it in.

File: loafwallet/CardView.swift (hunk 1 above)
Comment (mosadialiou) [reply loafwallet/CardView.swift:74]: hmmm I get it. I don't think the formatter will do the same. Can you put it in 1 line and apply the formatter. If you see a line break so you are right.

**Review by mosadialiou — APPROVED** [review]
LGTM

**Pull request merged**